core/banking-protocol/src/main/java/de/adorsys/opba/core/protocol/repository/jpa/BankProfileRepository.java

```diff
@@ -0,0 +1,11 @@
+package de.adorsys.opba.core.protocol.repository.jpa;
+
+import de.adorsys.opba.core.protocol.domain.entity.BankProfile;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface BankProfileRepository extends JpaRepository<BankProfile, Long> {
+
+    @Query("FROM BankProfile bp WHERE bp.bank.id = ?1")
+    BankProfile getProfileByBankId(Long bankId);
```

Comment:
Named query looks better, also better use Optional for REST-facing queries as it is clearer for exception handling.
Optional<BankProfile> findByBankId(Long bankId) that can be returned by endpoint too

Reply:
ok